src/app/clusters/smoke-co-alarm-server/SmokeCOTestEventTriggerDelegate.h

```diff
@@ -0,0 +1,83 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/TestEventTriggerDelegate.h>
+
+namespace chip {
+
+enum class SmokeCOTrigger : uint64_t
```

Comment:
Do we have some form of encoding specification for these events?

I tried to look at 7.18.2 for MEI encoding and that only covers 32-bit.

I am wondering what guarantees that `....00090` and up does not clash with other triggers. How are we setting these up?

Reply:
Maybe you want to see this. The EventTrigger with a value of 0xFFFF_FFFF_* is determined by the manufacturer and can be prevented from conflicting using the EnableKey.
[src/service_device_management/DiagnosticsGeneral.adoc](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/259fe2ea48221b9d55916acefe8f9c1fb7aa017d/src/service_device_management/DiagnosticsGeneral.adoc#eventtrigger-field)

Reply:
Thank you, that is the doc I was looking for